pkg/openwsn/contrib/eui64.c

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (C) 2017 Hamburg University of Applied Sciences
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     pkg_openwsn
+ * @{
+ *
+ * @file
+ * @brief       RIOT adaption definition of the "eui64" bsp module
+ *
+ * @author      Peter Kietzmann <peter.kietzmann@haw-hamburg.de>
+ *
+ * @}
+ */
+
+#include "string.h"
+#include "eui64.h"
+
+#include "luid.h"
+#include "net/netdev.h"
+#include "net/netopt.h"
+#include "net/ieee802154.h"
+
+void eui64_get(uint8_t *addressToWrite)
+{
+    /* this will be overridden in the bootstrapping function */
```

Comment:
I described this in https://github.com/RIOT-OS/RIOT/pull/13824#issuecomment-630679728, but maybe I could add a more detailed comment in code.

So OpenWSN recovers the eui64 and then generates a 16bit ID from that, but that ID was not unique at all for `iotla-m3`, allways `1114`, so although I provide this function I actually override the short ID and eui64 definitions in `openwsn_bootstrap()` to get the same identifiers that we use in RIOT (and in the process fix (or attenuate) the collision issue).

Reply:
But this is just the eui *get* function, not the one that sets the ID so the information should be irrelevant here, right?

Reply:
> But this is just the eui get function, not the one that sets the ID so the information should be irrelevant here, right?

Yes but in OpenWSN `id_manager` during `init` it calls `eui64_get` and sets `idmanager_vars.my64bID` then from that it sets `idmanager_vars.my16bID`. But I overlooked that this is also called once in `cjoin` code.  And since this was also called in `cjoin` code which made join procedure fail. 

I'll return this section to how it was before.

Reply:
Thanks for insisting on this, helped me spot an issue!